Source/WebKit/UIProcess/Extensions/WebExtension.h

```diff
@@ -68,16 +74,23 @@ class WebExtension : public API::ObjectImpl<API::Object::Type::WebExtension>, pu
     WTF_MAKE_NONCOPYABLE(WebExtension);
 
 public:
+    using Resources = HashMap<String, std::span<const uint8_t>>;
+    using IconsCacheVariant = std::variant<RefPtr<WebCore::Icon>, HashSet<String>>;
+    using IconsCache = HashMap<String, IconsCacheVariant>; // Either store image, or list of scales
+
     template<typename... Args>
     static Ref<WebExtension> create(Args&&... args)
     {
         return adoptRef(*new WebExtension(std::forward<Args>(args)...));
     }
 
-    explicit WebExtension(NSBundle *appExtensionBundle, NSError ** = nullptr);
-    explicit WebExtension(NSURL *resourceBaseURL, NSError ** = nullptr);
-    explicit WebExtension(NSDictionary *manifest, NSDictionary *resources);
-    explicit WebExtension(NSDictionary *resources);
+#if PLATFORM(COCOA)
+    explicit WebExtension(NSBundle *appExtensionBundle);
+#endif
+    explicit WebExtension(const JSON::Value& manifest, const Resources&);
+    explicit WebExtension(const Resources&);
```

Comment:
Should make these so they can be r-value references so the data can be moved.

```suggestion
    explicit WebExtension(const JSON::Value& manifest, const Resources&&);
    explicit WebExtension(const Resources&&);
```